### PR TITLE
feat(mux): Add Mux Video example

### PR DESCRIPTION
**Pull request by decepulis**
Ahoy! Adding a [Mux Video](https://www.mux.com) example following a basic UGC upload workflow. You can see a similar flow implemented at https://stream.new.

Looking forward to any feedback, and looking forward to getting this example into the hands of Remix devs!

### CODE REVIEW

**Review by machour — CHANGES_REQUESTED** [review]
Hey 👋 

Thank you so much for the contribution and sorry for the delays getting back to you. 

I'd like to get this example merged, but I wonder if you could: 
- upgrade the template to the Vite version (see the updated __template folder) 
- remove tailwindcss usage (to keep the example minimalistic) 
- drop the pricing part from the README, as a user interested in Mux will get all that info from the official website

Let me know if you have thoughts about this 🙏

**Comment by decepulis** [comment]
Hey @machour! Thanks for the feedback. It's addressed.

**Comment by decepulis** [comment]
Ah linting errors. Taking a look...

File: mux-video/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Open this example on [CodeSandbox](https://codesandbox.com):
+
+[![Open in CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/remix-run/examples/tree/main/mux-video)
+
+## How to use
+
+### Step 1. Create a Remix app with this example
+
+```bash
+npx create-remix@latest --template examples/mux-video
```
Comment (machour) [inline mux-video/README.md:22]:
```suggestion
npx create-remix@latest --template remix-run/examples/mux-video
```

File: mux-video/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+### Step 3. Set up environment variables
+
+Copy the `.env.example` file in this directory to `.env` (which will be ignored by Git):
+
+```bash
+cp .env.example .env
+```
+
+Then, go to the [settings page](https://dashboard.mux.com/settings/access-tokens) in your Mux dashboard, get a new **API Access Token**. Use that token to set the variables in `.env.local`:
```
Comment (machour) [inline mux-video/README.md:39]: Can you specify the permissions needed ? (I ticked everything 😅 )

**Review by machour — APPROVED** [review]
Neat work, thank you!

Just two little details in the README and we're good to go!

By the way, tested the example locally, uploaded a video, and when it was done, I had an error flash before the video got rendered. Are you getting the same thing? (I'll suspect my local env otherwise)

Screenshot 2024-05-30 at 3 38 30 PM

**Comment by decepulis** [comment]
I'm seeing the same rule of hooks error. But only the first time we run into that page, and only if that page isn't the first page I visit on boot. Similarly, if the first page I visit is the `playback` page, `mux-player-react` loads fine but `mux-uploader-react` throws a rule of hooks error. The errors go away after I _hard_ reload the page, clearing the browser cache. The errors re-appear if I clear my `node_modules` and start again.

I wonder if it's related to `[vite] ✨ new dependencies optimized: @mux/mux-player-react`; I only notice the error before that happens.

I don't know enough about vite or how it optimizes dependencies to be sure, though. Do you?

**Pull request merged**